File: CS/DocumentPrintPreview/ViewModel.cs
using DevExpress.DataAccess.Sql;
using DevExpress.DataAccess.Sql.DataApi;
using DevExpress.Mvvm.POCO;
using System.ComponentModel;

namespace DocumentPrintPreview {
    public class ViewModel {
        public object DataSource { get; }
        public virtual EmployeeReport Report { get; protected set; }
        public virtual IRow SelectedEmployee { get; set; }
        public CustomSettingsViewModel CustomSettings { get; }

        protected ViewModel() {
            CustomSettings = ViewModelSource.Create(() => new CustomSettingsViewModel());
            ((INotifyPropertyChanged)CustomSettings).PropertyChanged += OnCustomSettingsChanged;
            DataSource = CreateDataSource();
        }

        void OnCustomSettingsChanged(object sender, PropertyChangedEventArgs e) {
            if(Report!=null) {
                Report.Parameters["ShowSubordinates"].Value = CustomSettings.ShowSubordinates;
                Report.CreateDocument(true);
            }
        }

        protected void OnSelectedEmployeeChanged() {
            var id = SelectedEmployee.GetValue<long>("EmployeeID");
            var report = new EmployeeReport();
            report.Parameters["EmployeeId"].Value = id;
            report.Parameters["ShowSubordinates"].Value = CustomSettings.ShowSubordinates;
            report.DataSource = DataSource;
            Report = report;
        }

        SqlDataSource CreateDataSource() {
            var sqlDataSource = new SqlDataSource();
            sqlDataSource.ConnectionName = "nwind";
            var query = SelectQueryFluentBuilder
                .AddTable("Employees")
                .SelectAllColumns()
                .Build("Employees");
            var masterDetailInfo = new MasterDetailInfo("Employees", "Employees", "EmployeeID", "ReportsTo");
            sqlDataSource.Queries.Add(query);
            sqlDataSource.Relations.Add(masterDetailInfo);
            sqlDataSource.Fill();
            return sqlDataSource;
        }
    }

    public class CustomSettingsViewModel {
        public virtual bool ShowSubordinates { get; set; }
        public CustomSettingsViewModel() {
            ShowSubordinates = true;
        }
    }
}

File: VB/DocumentPrintPreview/ViewModel.vb
Imports DevExpress.DataAccess.Sql
Imports DevExpress.DataAccess.Sql.DataApi
Imports DevExpress.Mvvm.POCO
Imports System.ComponentModel

Namespace DocumentPrintPreview

    Public Class ViewModel

        Public ReadOnly Property DataSource As Object

        Public Overridable Property Report As EmployeeReport

        Public Overridable Property SelectedEmployee As IRow

        Public ReadOnly Property CustomSettings As CustomSettingsViewModel

        Protected Sub New()
            CustomSettings = ViewModelSource.Create(Function() New CustomSettingsViewModel())
            AddHandler CType(CustomSettings, INotifyPropertyChanged).PropertyChanged, AddressOf OnCustomSettingsChanged
            DataSource = CreateDataSource()
        End Sub

        Private Sub OnCustomSettingsChanged(ByVal sender As Object, ByVal e As PropertyChangedEventArgs)
            If Report IsNot Nothing Then
                Report.Parameters("ShowSubordinates").Value = CustomSettings.ShowSubordinates
                Report.CreateDocument(True)
            End If
        End Sub

        Protected Sub OnSelectedEmployeeChanged()
            Dim id = SelectedEmployee.GetValue(Of Long)("EmployeeID")
            Dim report = New EmployeeReport()
            report.Parameters("EmployeeId").Value = id
            report.Parameters("ShowSubordinates").Value = CustomSettings.ShowSubordinates
            report.DataSource = DataSource
            Me.Report = report
        End Sub

        Private Function CreateDataSource() As SqlDataSource
            Dim sqlDataSource = New SqlDataSource()
            sqlDataSource.ConnectionName = "nwind"
            Dim query = SelectQueryFluentBuilder.AddTable("Employees").SelectAllColumns().Build("Employees")
            Dim masterDetailInfo = New MasterDetailInfo("Employees", "Employees", "EmployeeID", "ReportsTo")
            sqlDataSource.Queries.Add(query)
            sqlDataSource.Relations.Add(masterDetailInfo)
            sqlDataSource.Fill()
            Return sqlDataSource
        End Function
    End Class

    Public Class CustomSettingsViewModel

        Public Overridable Property ShowSubordinates As Boolean

        Public Sub New()
            ShowSubordinates = True
        End Sub
    End Class
End Namespace
